model/src/main/java/io/paradoxical/cassieq/model/auth/SignatureGenerator.java

```diff
@@ -0,0 +1,43 @@
+package io.paradoxical.cassieq.model.auth;
+
+import com.godaddy.logging.Logger;
+import io.paradoxical.cassieq.model.accounts.AccountKey;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import static com.godaddy.logging.LoggerFactory.getLogger;
+
+/**
+ * Signs paramters
+ */
+public interface SignatureGenerator {
+
+    default String computeSignature(AccountKey key) throws InvalidKeyException, NoSuchAlgorithmException {
```

Comment:
honestly we should swallow these throws as well, since HmacSHA256 is a required algorithm for all java implementations. Invalid key shouldn't be possible either, since we created the key and SecretKeySpec IS the correct class (not user supplied)

if they do happen, we should raise an `AssertionError`